src/codeExecutionWorker.js

```diff
@@ -0,0 +1,67 @@
+/* global eval */
+'use strict'
+import { connectIframe, PluginClient } from '@remixproject/plugin'
+import ethers from 'ethers' // eslint-disable-line
+import Web3 from 'web3' // eslint-disable-line
+import swarmgw_fn from 'swarmgw' // eslint-disable-line
+window.swarmgw = swarmgw_fn() // eslint-disable-line
+window.ethers = ethers
+window.Web3 = Web3
+
+class Web3Provider {
+  constructor (plugin) {
+    this.plugin = plugin
+  }
+  sendAsync (payload, callback) {
+    this.plugin.call('appWeb3Provider', 'sendAsync', payload).then(result => callback(null, result)).catch(e => callback(e.message))
+  }
+}
+
+class CodeExecutor extends PluginClient {
+  execute (script) {
+    if (!this.currentRequest.isFromNative) return
+    if (script) { // looks like this was called more than one time, the second time script was undefined...  might be a bug in remix-plugin
+      try {
+        (new Function(script))() // eslint-disable-line
+      } catch (e) {
+        this.emit('log', {
+          cmd: 'error',
+          data: [e.message]
+        })
+      }
+    }
+  }
+}
+
+window.codeExec = new CodeExecutor()
+window.appWeb3Provider = new Web3Provider(window.codeExec)
```

Comment:
Do you really need this ? I would consider interacting with the provider through web3 or ethers.

Reply:
developer can then use instanciate ethers themselves

Reply:
Which developers ? I don't understand who is supposed to have access to this "window" object.
Also, why web3 is defined by default and not ethers ? I think you want to expose both or none, don't you think ?

Reply:
> why web3 is defined by default and not ethers 

that was the case before, but I can change that to expose both

Reply:
> Which developers

Remix IDE users
use case is: you put some javascript code in a file in Remix and run `remix.execute()`.

Reply:
So you need to put the values in the window to be able to run this command ? So they don't have to do `const Web3 = await import('https://unpkg.com/web3')` ?

Reply:
> So you need to put the values in the window to be able to run this command ?

when you run `remix.execute()`, the plugin just execute the code from the selected javascript file. that has nothing to do with the values in `window`.
we need anyway to inject the provider, because it is a plugin under the hood.
Then web3.js and ether are directly injected in the plugin scoped, so user can use them from the script.

Reply:
So if you don't need to access the values in the window object, why do you set them ?

Reply:
> So you need to put the values in the window to be able to run this command 

you can run the command even if there's nothing in the window object.
having something in the window object is not mandatory for running the script.
If you don't have web3 in the window object then you can use it from your script.